base/sparse/sparsematrix.jl

```diff
@@ -17,12 +17,16 @@ immutable SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
         n < 0 && throw(ArgumentError("number of columns (n) must be ≥ 0, got $n"))
         new(Int(m), Int(n), colptr, rowval, nzval)
     end
+    SparseMatrixCSC(m::Integer, n::Integer) = SparseMatrixCSC{Tv,Ti}(m, n, ones(Ti, max(n,0)+1), Array{Ti}(0), Array{Tv}(0))
```

Comment:
this should probably check for negative n instead of allowing n not to match length(colptr)+1